sdk/core/azure-core/src/main/java/com/azure/core/util/polling/DefaultSyncPoller.java

```diff
@@ -206,6 +206,17 @@ public void cancelOperation() {
         }
     }
 
+    @Override
+    public SyncPoller<T, U> setPollInterval(Duration pollInterval) {
+        Objects.requireNonNull(pollInterval, "'pollInterval' cannot be null.");
+        if (pollInterval.compareTo(Duration.ZERO) <= 0) {
```

Comment:
Not sure if calls to both `Duration.isZero()` and `Duration.isNegative()` would be more performant but it would be an easier read

Reply:
Agree that `pollInterval.isZero() || pollInterval.isNegative()` is more readable. Updated the code.